rwengine/src/ai/CharacterControllerActivities.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#ifndef _CONTROLLER_ACTIVITY_HPP_
+#define _CONTROLLER_ACTIVITY_HPP_
+
+#include <ai/CharacterController.hpp>
+
+
+#define DECL_ACTIVITY(act_name)                                               \
+	class act_name : public CharacterController::Activity                 \
+	{                                                                     \
+	public:                                                               \
+		static constexpr auto ActivityName = #act_name;               \
+		std::string name() const override { return ActivityName; };   \
+		bool update(CharacterObject*, CharacterController*) override;
+
+#define DECL_ACTIVITY_CANSKIP(act_name)                          \
+	DECL_ACTIVITY(act_name)                                  \
+		bool canSkip(CharacterObject *) const override;
+
+
+class CharacterObject;
+class VehicleObject;
+class WeaponItem;
+
+
+namespace Activities
+{
+	DECL_ACTIVITY_CANSKIP(GoTo)
+		GoTo(const glm::vec3& target, bool sprint=false) :
+			target(target), sprint(sprint) { }
+
+	private:
+		glm::vec3 target;
+		bool sprint;
+	};
+
+
+	DECL_ACTIVITY(Jump)
+		Jump() : jumped(false) { }
+
+	private:
+		bool jumped;
+	};
+
+
+	DECL_ACTIVITY_CANSKIP(EnterVehicle)
+		enum {
+			ANY_SEAT = -1 // Magic number for any seat but the driver's.
+		};
+
+		EnterVehicle(VehicleObject* vehicle, int seat=0) :
+			vehicle(vehicle), seat(seat), entering(false) { }
+
+	private:
+		VehicleObject* vehicle;
+		int seat;
+		bool entering;
+	};
+
+
+	DECL_ACTIVITY(ExitVehicle)
+		ExitVehicle(bool jacked=false) : jacked(jacked) { }
+
+	private:
+		const bool jacked;
+	};
+
+
+	DECL_ACTIVITY(ShootWeapon)
+		ShootWeapon(WeaponItem* item) : item(item), fired(false) { }
+
+	private:
+		WeaponItem* item;
+		bool fired;
```

Comment:
I believe the authors intention for prefixing those with `_` in the old code is because they are member variables? (so `m_` would have been more appropriate)
Anyway, your solution is fine because it improves consistency within this file.